server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstractionResolver.java

```diff
@@ -103,7 +103,7 @@ public IndexAbstractionResolver(IndexNameExpressionResolver indexNameExpressionR
             } else if (dateMathName.equals(indexAbstraction)) {
```

Comment:
Note that the current behavior in Security is different the "core" behavior.
In core, excluding missing index names (ie "-nonexistent") is an error, but not in this case.
Most likely, the fix for it is to adopt the same behavior from the date math expression evaluation from a couple of lines above.

But, I'll stay away from it for now.
The robust, non-patchy fix, is to inject into the index name resolver from core the authorized indices view.

The proposed fix in this PR maintains the current (buggy) behavior, but avoids redundant `availableIndexAbstractions.contains` checks, hopefully having a small positive performance impact.